### PR TITLE
Fixes #10 - Update helloworld example to new Connector builder

### LINKED ISSUE #10: Update hello world example to use new Connector builder

### PR DESCRIPTION
**Pull request by mikewalch**
* Example also now uses conf/accumulo-client.properties
* Added a log4j.properties file for examples

I will change the README.md to tell users to configure `accumulo-client.properties`.  I wanted to get feedback first on these changes to helloworld before I make them to other examples and update the docs.

### CODE REVIEW

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java [hunk 1]
```diff
@@ -20,46 +20,42 @@
...
+
+  public static void main(String[] args) throws AccumuloException, AccumuloSecurityException, TableExistsException,
       TableNotFoundException {
-    ClientOnRequiredTable opts = new ClientOnRequiredTable();
-    BatchWriterOpts bwOpts = new BatchWriterOpts();
-    opts.parseArgs(InsertWithBatchWriter.class.getName(), args, bwOpts);
 
-    Connector connector = opts.getConnector();
-    MultiTableBatchWriter mtbw = connector.createMultiTableBatchWriter(bwOpts.getBatchWriterConfig());
+    Connector connector = Connector.builder().usingProperties("conf/accumulo-client.properties").build();
```
Comment (milleruntime) [inline src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: +1 Not only using the new API but using a more realistic example of how to connect.

File: src/main/java/org/apache/accumulo/examples/helloworld/ReadData.java [hunk 2]
```diff
@@ -27,52 +26,29 @@
...
-    Iterator<Entry<Key,Value>> iter = scan.iterator();
+    Connector connector = Connector.builder().usingProperties("conf/accumulo-client.properties").build();
 
-    while (iter.hasNext()) {
-      Entry<Key,Value> e = iter.next();
-      Text colf = e.getKey().getColumnFamily();
-      Text colq = e.getKey().getColumnQualifier();
-      log.trace("row: " + e.getKey().getRow() + ", colf: " + colf + ", colq: " + colq);
-      log.trace(", value: " + e.getValue().toString());
+    try (Scanner scan = connector.createScanner("hellotable", Authorizations.EMPTY)) {
```
Comment (milleruntime) [inline src/main/java/org/apache/accumulo/examples/helloworld/ReadData.java:44]: +1 Modern use of scanner.

**Review by milleruntime — APPROVED** [review]
Looks good.  I think there is a lot more we could do to make examples more user friendly for newcomers but this is a good start.

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java [hunk 3]
```diff
@@ -20,46 +20,42 @@
...
-    System.out.println("writing ...");
-    for (int i = 0; i < 10000; i++) {
-      Mutation m = new Mutation(new Text(String.format("row_%d", i)));
-      for (int j = 0; j < 5; j++) {
-        m.put(colf, new Text(String.format("colqual_%d", j)), new Value((String.format("value_%d_%d", i, j)).getBytes()));
+    try (BatchWriter bw = connector.createBatchWriter("hellotable")) {
+      Text colf = new Text("colfam");
+      log.trace("writing ...");
+      for (int i = 0; i < 10000; i++) {
+        Mutation m = new Mutation(new Text(String.format("row_%d", i)));
```
Comment (keith-turner) [inline src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: Can now pass Strings to Mutation, used to could not.  Changing this could be a separate PR.

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java [hunk 4]
```diff
@@ -20,46 +20,42 @@
...
 
-    Connector connector = opts.getConnector();
-    MultiTableBatchWriter mtbw = connector.createMultiTableBatchWriter(bwOpts.getBatchWriterConfig());
+    Connector connector = Connector.builder().usingProperties("conf/accumulo-client.properties").build();
 
-    if (!connector.tableOperations().exists(opts.getTableName()))
-      connector.tableOperations().create(opts.getTableName());
-    BatchWriter bw = mtbw.getBatchWriter(opts.getTableName());
+    if (!connector.tableOperations().exists("hellotable")) {
+      connector.tableOperations().create("hellotable");
```
Comment (keith-turner) [inline src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: A better pattern for this is to call create and catch a table exists exception.  Can do nothing when the exception is caught. Calling exists and then create is susceptible to race conditions.  This could be changed in another PR.

File: src/main/java/org/apache/accumulo/examples/helloworld/ReadData.java [hunk 5]
```diff
@@ -27,52 +26,29 @@
...
-  }
-
   public static void main(String[] args) throws AccumuloException, AccumuloSecurityException, TableNotFoundException {
-    Opts opts = new Opts();
-    ScannerOpts scanOpts = new ScannerOpts();
-    opts.parseArgs(ReadData.class.getName(), args, scanOpts);
-
-    Connector connector = opts.getConnector();
 
-    Scanner scan = connector.createScanner(opts.getTableName(), opts.auths);
```
Comment (keith-turner) [inline src/main/java/org/apache/accumulo/examples/helloworld/ReadData.java:59]: Really nice to see so much code go away.

File: src/main/java/org/apache/accumulo/examples/helloworld/ReadData.java [hunk 6]
```diff
@@ -27,52 +26,29 @@
...
-    scan.setBatchSize(scanOpts.scanBatchSize);
-    Key start = null;
-    if (opts.startKey != null)
-      start = new Key(new Text(opts.startKey));
-    Key end = null;
-    if (opts.endKey != null)
-      end = new Key(new Text(opts.endKey));
-    scan.setRange(new Range(start, end));
-    Iterator<Entry<Key,Value>> iter = scan.iterator();
+    Connector connector = Connector.builder().usingProperties("conf/accumulo-client.properties").build();
```
Comment (keith-turner) [inline src/main/java/org/apache/accumulo/examples/helloworld/ReadData.java:42]: When I first saw this I was thinking "maybe the props file could be args[0]".  However I convinced myself that the literal string is better because its easier to understand.

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java [hunk 7]
```diff
@@ -20,46 +20,42 @@
...
+      Text colf = new Text("colfam");
+      log.trace("writing ...");
+      for (int i = 0; i < 10000; i++) {
+        Mutation m = new Mutation(new Text(String.format("row_%d", i)));
+        for (int j = 0; j < 5; j++) {
+          m.put(colf, new Text(String.format("colqual_%d", j)), new Value((String.format("value_%d_%d", i, j)).getBytes()));
+        }
+        bw.addMutation(m);
+        if (i % 100 == 0)
+          log.trace(String.valueOf(i));
```
Comment (keith-turner) [inline src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: Why log at trace level?  Why use logger instead of System.out?  It seems like if we want to make the examples easy to understand then using System.out would be better than logging.

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java (hunk 3 above)
Comment (mikewalch) [reply src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: I will update the PR

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java (hunk 7 above)
Comment (mikewalch) [reply src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: I want users to be able to the class doing the logging.  Also, logging is now automatically set up for the user so there is no risk they won't see these messages.

File: src/main/java/org/apache/accumulo/examples/helloworld/ReadData.java (hunk 6 above)
Comment (mikewalch) [reply src/main/java/org/apache/accumulo/examples/helloworld/ReadData.java:42]: It would also make argument parsing tricky.

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java (hunk 4 above)
Comment (mikewalch) [reply src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: I will update PR.  I also created #13

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java (hunk 3 above)
Comment (mikewalch) [reply src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: Also created #12

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java (hunk 4 above)
Comment (mikewalch) [reply src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: Fixed in 2ac71cb726f

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java (hunk 3 above)
Comment (mikewalch) [reply src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: Fixed in 2ac71cb726f

File: src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java (hunk 4 above)
Comment (milleruntime) [reply src/main/java/org/apache/accumulo/examples/helloworld/InsertWithBatchWriter.java]: I think it would be good to include this race condition in the Accumulo API somehow.  Perhaps we could create an overloaded method like we did with NewTableConfig.  Maybe a boolean flag that won't throw an exception if the table already exists.  Just a way to handle the "do nothing" exception use case.

**Review by keith-turner — APPROVED** [review]
